base_layer/key_manager/src/cipher_seed.rs

```diff
@@ -264,6 +281,92 @@ impl CipherSeed {
     }
 }
 
+impl CipherSeed {
```

Comment:
Using the `MacDomain` from the hasing API:
```suggestion
    // TODO: Remove when generic `HashingDomain::mac_hasher` implementation is ready
    fn mac_hasher<D: Digest + LengthExtensionAttackResistant>(
        domain_label: &'static str,
    ) -> DomainSeparatedHasher<D, MacDomain> {
        DomainSeparatedHasher::<D, MacDomain>::new(domain_label)
    }

    fn generate_mac(
```

Reply:
Great suggestion !